stl/inc/xstring

```diff
@@ -4001,22 +3997,21 @@ public:
             return;
         }
 
-        if (_My_data._Mysize < _BUF_SIZE) {
+        if (!_Entails_large_string(_My_data._Mysize)) {
             _Become_small();
             return;
         }
 
-        const size_type _Target_capacity = (_STD min)(_My_data._Mysize | _ALLOC_MASK, max_size());
+        size_type _Target_capacity = (_STD min)(_My_data._Mysize | _ALLOC_MASK, max_size());

```

Comment:
(no changes requested) As `_Exactly` mode doesn't really require input capacity to be a referrence, I think we can restore constness for `_Target_capacity` in the future. Also here in `<vector>`:
https://github.com/microsoft/STL/blob/0fe653a9d6efd2ac0ad07825804436672dbffe3b/stl/inc/vector#L1705-L1706